src/test/kotlin/org/rust/ide/commenter/RsCommenterTest.kt

```diff
@@ -147,4 +242,62 @@ class RsCommenterTest : RsTestBase() {
         <caret>
          */
     """, IdeActions.ACTION_EDITOR_ENTER)
+
+    fun `test move caret after commenting empty line`() = checkOption(settings()::LINE_COMMENT_ADD_SPACE, """
+        <caret>
+        fn foo() {}
+    """, """
+        // <caret>
+        fn foo() {}
+    """, """
+        //<caret>
+        fn foo() {}
+    """, IdeActions.ACTION_COMMENT_LINE)
+
+    private fun checkOption(optionProperty: KMutableProperty0<Boolean>,
+                            @Language("Rust") before: String,
+                            @Language("Rust") afterOn: String,
+                            @Language("Rust") afterOff: String,
+                            actionId: String,
+                            trimIndent: Boolean = true) {
+        val initialValue = optionProperty.get()
+        optionProperty.set(true)
+        try {
+            checkEditorAction(before, afterOn, actionId, trimIndent = trimIndent)
+            resetActionManagerState()
+            optionProperty.set(false)
+            checkEditorAction(before, afterOff, actionId, trimIndent = trimIndent)
+            resetActionManagerState()
+        } finally {
+            optionProperty.set(initialValue)
+        }
+    }
+
+    private fun settings() = CodeStyle.getSettings(project).getCommonSettings(RsLanguage)
+
+    /**
+     * Resets [com.intellij.openapi.actionSystem.impl.ActionManagerImpl.myPrevPerformedActionId].
+     * Otherwise it affect [com.intellij.codeInsight.generation.CommentByLineCommentHandler.invoke]
+     * (see `startingNewLineComment` condition there).
+     */
+    private fun resetActionManagerState() {
+        myFixture.performEditorAction(EMPTY_ACTION_ID)
+    }
+
+    override fun setUp() {
+        super.setUp()
+        ActionManager.getInstance().registerAction(EMPTY_ACTION_ID, EmptyAction.createEmptyAction("empty", null, true))
+    }
+
+    override fun tearDown() {
+        try {
+            super.tearDown()
+        } finally {
+            ActionManager.getInstance().unregisterAction(EMPTY_ACTION_ID)
+        }
+    }
```

Comment:
```suggestion
    override fun tearDown() {
        ActionManager.getInstance().unregisterAction(EMPTY_ACTION_ID)
        super.tearDown()
    }
```
It's recommended way to cleanup our custom state before `super.tearDown()`.

Alternatively, you can drop `tearDown` at all and add the following code in `setUp`:
```kotlin
    override fun setUp() {
        super.setUp()
        val actionManager = ActionManager.getInstance()
        actionManager.registerAction(EMPTY_ACTION_ID, EmptyAction.createEmptyAction("empty", null, true))
        Disposer.register(testRootDisposable, Disposable {
            actionManager.unregisterAction(EMPTY_ACTION_ID)
        })
    }
```

Reply:
Thanks, let's see if this appeases CI.